src/evo/mnhftx.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2021-2022 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <evo/mnhftx.h>
+#include <llmq/commitment.h>
+#include <llmq/signing.h>
+#include <evo/specialtx.h>
+#include <consensus/validation.h>
+
+#include <chain.h>
+#include <chainparams.h>
+#include <logging.h>
+#include <validation.h>
+
+extern const std::string CBLSIG_REQUESTID_PREFIX = "clsig";
+
+bool MNHFTx::Verify(const CBlockIndex* pQuorumIndex) const
+{
+    if (nVersion == 0 || nVersion > CURRENT_VERSION) {
+        return false;
+    }
+
+    Consensus::LLMQType llmqType = Params().GetConsensus().llmqTypeMnhf;
+    int signOffset{llmq::GetLLMQParams(llmqType).dkgInterval};
+
+    const uint256 requestId = ::SerializeHash(std::make_pair(CBLSIG_REQUESTID_PREFIX, pQuorumIndex->nHeight));
+    if(llmq::quorumSigningManager->VerifyRecoveredSig(llmqType, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, 0) ||
+            llmq::quorumSigningManager->VerifyRecoveredSig(llmqType, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, signOffset)) {
+        if (!sig.IsValid()) {
+           return false;
+        }
+    }
+
+    return true;
```

Comment:
This looks wrong... shouldn't it be simply
```suggestion
return llmq::quorumSigningManager->VerifyRecoveredSig(llmqType, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, 0) ||
            llmq::quorumSigningManager->VerifyRecoveredSig(llmqType, pQuorumIndex->nHeight, requestId, pQuorumIndex->GetBlockHash(), sig, signOffset);
```
?

Reply:
agree and code updated